src/helpers/yjs.js

```diff
@@ -42,3 +55,29 @@ export function applyDocumentState(ydoc, documentState, origin) {
 	const update = decodeArrayBuffer(documentState)
 	applyUpdate(ydoc, update, origin)
 }
+
+/**
+ * @param {string} update - base64 encoded Y.js message
+ */
+export function getMessageType(update) {
+	const decoder = decoding.createDecoder(update)
+	return decoding.readVarUint(decoder)
+}
+
+/**
+ * @param {string} update - base64 encoded Y.js message
+ */
+export function getAwarenessMessageClientIds(update) {
+	const decoder = decoding.createDecoder(update)
+	const type = decoding.readVarUint(decoder)
+	if (type !== yjsMessageTypes.awareness) {
+		return []
+	}
+
+	const len = decoding.readVarUint(decoder)
+	const clientIds = []
+	for (let i = 0; i < len; i++) {
+		clientIds.push(decoding.readVarUint(decoder))
```

Comment:
I think we still need to read the other values here as well to continue at the correct byte position for the next client if.

Reply:
Good point. Tried to fix the code now. But given that I for now didn't see an update message with more than one awareness update, I was not able to test this in practice. Do you have a good idea how to create a test scenario?